grimoirelab_toolkit/credential_manager/bw_manager.py

```diff
@@ -0,0 +1,186 @@
+#
+#
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Author:
+#       Alberto Ferrer Sánchez (alberefe@gmail.com)
+#
+
+import json
+import subprocess
+import logging
+import os
+from typing import Any
+
+from .exceptions import (
+    BitwardenCLIError,
+    InvalidCredentialsError,
+    SessionNotFoundError,
+    CredentialNotFoundError,
+    ConnectionError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class BitwardenManager:
+    def __init__(self):
+        """
+        Logs in to Bitwarden using API key authentication from environment variables.
+
+        :raises InvalidCredentialsError: If invalid credentials are provided
+        :raises BitwardenCLIError: If Bitwarden CLI operations fail
+        :raises ConnectionError: If connection issues occur
+        """
+        # Session key of the bw session
+        self.session_key = None
+        self.formatted_credentials = {}
+
+        # Get API credentials from environment variables
+        self.client_id = os.getenv("BW_CLIENTID")
+        self.client_secret = os.getenv("BW_CLIENTSECRET")
+
+        if not self.client_id or not self.client_secret:
+            raise InvalidCredentialsError(
+                "BW_CLIENTID and BW_CLIENTSECRET environment variables must be set"
+            )
+
+        try:
+            self._login()
+        except (InvalidCredentialsError, BitwardenCLIError, ConnectionError) as e:
+            logger.error("Failed to initialize Bitwarden manager: %s", e)
+            raise
+
+    def _login(self) -> str | None:
```

Comment:
This method should be public. Also, make it explicit that the developer has to call it. In the same way we have an explicit log out, make the log in explicit too.